server/webhook.go

```diff
@@ -199,3 +202,79 @@ func (p *Plugin) GetWebhookURL(jiraURL string, teamID, channelID string) (subURL
 
 	return subURL, legacyURL, nil
 }
+
+func (wh *webhook) getConnection(p *Plugin, instance Instance, notification webhookUserNotification) (con *Connection, err error) {
+	var mattermostUserID types.ID
+
+	// prefer accountId to username when looking up UserIds
+	if notification.jiraAccountID != "" {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), notification.jiraAccountID)
+	} else {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), notification.jiraUsername)
+	}
+	if err != nil {
+		return
+	}
+
+	// Check if the user has permissions.
+	con, err = p.userStore.LoadConnection(instance.GetID(), mattermostUserID)
+	if err != nil {
+		// Not connected to Jira, so can't check permissions
+		return
+	}
+
+	return
+}
+
+func (wh *webhook) checkWatcherUser(p *Plugin, instance Instance) {
+	watcherUsers := &[]jira.User{}
+	for _, notification := range wh.notifications {
+		c, err := wh.getConnection(p, instance, notification)
+		if err != nil {
+			continue
+		}
+		client, err2 := instance.GetClient(c)
+		if err2 != nil {
+			continue
+		}
+		watcherUsers, err = client.GetWatchers(wh.Issue.ID)
+		if err != nil {
+			continue
+		}
+		break
+	}
```

Comment:
actually, the idea is to call `GetWatchers` from `client.Jira.Issue` we have to create a client. where to create a client requires mattermostUserID. 
https://github.com/mattermost/mattermost-plugin-jira/blob/36bcf271f93be5f241c3b0e212b38405662cd61a/server/webhook.go#L118

Because I don't have any idea how to get a connection if not use mattermostUserID, That's why to get mattermostUserID I try to loop wh.notifications to get 1 mattermostID that eligible to call `GetWatchers` . 
from that logic, if successfully calling `GetWatchers` it will exit the loop and continue to the next process validating the results of `GetWatchers`.

is it possible to use `wh.JiraWebhook.User` to get the mattermostUserID?
```
	c, err := wh.getConnection(p, instance, webhookUserNotification{
		jiraAccountID: wh.JiraWebhook.User.AccountID,
		jiraUsername:  wh.JiraWebhook.User.Name,
	})
	if err != nil {
		return
	}
	client, err2 := instance.GetClient(c)
	if err2 != nil {
		return
	}
	watcherUsers, err := client.GetWatchers(wh.Issue.ID)
	if err != nil {
		return
	}
```